src/main/java/com/parsing/parsers/pdf/parsing/DataRecognizer.java

```diff
@@ -0,0 +1,156 @@
+package com.parsing.parsers.pdf.parsing;
+
+import com.parsing.parsers.pdf.parsing.model.Column;
+import com.parsing.parsers.pdf.parsing.model.Row;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class DataRecognizer {
+
+    private final LotPDFResultService lotPDFResultService;
+    @Value("${laptop.models}")
+    private final String[] laptopModels;
+    private String model = null;
+    private Integer amount = 0;
+    private BigDecimal price = null;
+    private BigDecimal totalPrice = null;
+
+    public final boolean recognizeLotPDFResult(List<Row> rows) {
+        boolean result = false;
+        for (Row row : rows) {
+            if (isModelRow(row)) {
+                int modelColumnNumber = getModelColumnNumber(row);
+                List<String> amounts = new ArrayList<>();
+                List<String> prices = new ArrayList<>();
+                for (int x = modelColumnNumber; x < row.getColumns().size(); x++) {
+                    if (findAmount(row.getColumns().get(x).getParsingResult()) != null) {
+                        amounts.add(findAmount(row.getColumns().get(x).getParsingResult()));
+                    }
+                    if (findPrices(row.getColumns().get(x).getParsingResult()) != null) {
+                        prices.add(findPrices(row.getColumns().get(x).getParsingResult()));
+                    }
+                }
+                getAmount(amounts);
+                getPriceAndTotalPrice(prices);
+                result = saveItems();
+            }
+        }
+        return result;
+    }
+
+    private boolean saveItems() {
+        if (totalPrice != null && totalPrice.equals(BigDecimal.valueOf(amount).multiply(price))) {
```

Comment:
consider creating a method for this statement

Reply:
Done.